pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go

```diff
@@ -5,11 +5,29 @@ import (
 	"regexp"
 )
 
+var validMetadataResources = map[string]map[string]bool{
```

Comment:
It's possible to use `DeviceFields` and `InterfaceTypes`, but downside might be making other part of the code less generic.

Let me check if I can do something about making this part easier to maintain.

Reply:
@FlorianVeaux Just thought about this, I think having dedicated types for DeviceFields and InterfaceTypes will make other part of the code less generic (e.g. metadata config struct and parser metadata store). 

I think it should be not that hard to maintain since when adding a new field, the validator will raise error very earlier on as soon as we try to use a new field that is not in the validator list.

We can discuss about it if you think there is a better way :)

Reply:
That's fine thanks for checking the feasibility of the idea. Let's not overcomplicate the design, and it's good to keep that part of the code generic enough for adding potential new resources in the future 👍